### PR TITLE
Fix bug with get dependants filtered caching its filter

### PR DESCRIPTION
**Pull request by ThetaSinner**
### Summary

Fixes a bug where the `filter_fn` used by `dependants_in_workspace_filtered` is cached by the `OnceCell` that it uses to avoid repeating work.

The caching has been moved up a level into `dependants_in_workspace` which always give back an unfiltered result. The filter will always recompute if called.

Discovered during https://github.com/holochain/holochain/pull/2149/files#diff-f0903ab3d30b02d95cbe6b4e77ee4d38e5b404f2346045990c357dc6a6f15f9fR354

### TODO:
- [x] CHANGELOG(s) updated with appropriate info
- [ ] Just before pressing the merge button, ensure new entries to CHANGELOG(s) are still under the _UNRELEASED_ heading 
- [ ] backport to develop-0.1

### CODE REVIEW

**Comment by ThetaSinner** [comment]
@steveeJ should this be targeting the `develop-0.1` branch?

File: crates/release-automation/src/lib/crate_selection/tests/mod.rs [hunk 1]
```diff
@@ -86,7 +86,7 @@ fn release_selection() {
 }
 
 #[test]
-fn members_dependencies() {
```
Comment (steveeJ) [inline crates/release-automation/src/lib/crate_selection/tests/mod.rs:89]: i think i should have not removed this in #2177. i was mistaken and thought you had introduced this test. there might be an opportunity to simplify that test but it doesn't have to be taken now. let me know if you prefer if i make another PR to your PR to bring this one back.

Comment (steveeJ) [reply crates/release-automation/src/lib/crate_selection/tests/mod.rs:89]: https://github.com/holochain/holochain/pull/2178

Comment (ThetaSinner) [reply crates/release-automation/src/lib/crate_selection/tests/mod.rs:89]: The tests have failed (flake) so I'll push a new commit with this test restored

Comment (ThetaSinner) [reply crates/release-automation/src/lib/crate_selection/tests/mod.rs:89]: Sorry, page wasn't refreshed, you got to it already. Thank you!

**Review by steveeJ — APPROVED** [review]
awesome, thanks for getting this fixed and tested :partying_face:
